src/components/ContentNode/SectionTitle.vue

```diff
@@ -0,0 +1,86 @@
+<!--
+  This source file is part of the Swift.org open source project
+
+  Copyright (c) 2022 Apple Inc. and the Swift project authors
+  Licensed under Apache License v2.0 with Runtime Library Exception
+
+  See https://swift.org/LICENSE.txt for license information
+  See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+-->
+
+<template>
+  <component
+   :id="!isTargetIDE ? id : null"
+   :is="tag"
+   class="section-title"
+  >
+    <a
+      v-if="anchor && !isTargetIDE"
+      :href="`#${anchor}`"
+      class="header-anchor"
+      aria-label="hidden"
+      @click="handleFocusAndScroll(anchor)"
+    >#</a>
+    <slot />
+  </component>
+</template>
+
+<script>
+import scrollToElement from 'docc-render/mixins/scrollToElement';
+
+export default {
+  name: 'SectionTitle',
+  mixins: [scrollToElement],
+  data() {
+    return {
+      id: null,
+    };
+  },
+  props: {
+    anchor: {
+      type: String,
+      required: false,
+    },
+    tag: {
+      type: String,
+      default: () => 'h2',
+    },
+  },
+  inject: {
+    isTargetIDE: {
+      default: () => false,
+    },
+  },
+  mounted() {
+    if (!this.anchor) return;
+    const element = document.getElementById(`${this.anchor}`);
```

Comment:
```suggestion
    const element = document.getElementById(this.anchor);
```

As a more general comment though, I don't totally understand the need for why this hook is needed. Why are there separate `id` and `anchor` values and how are they used differently?

Reply:
There are cases where the ID is applied on a section that wraps the heading (like in TopicsSection), but in other cases, the heading has to hold the ID.

Reply:
I'm not totally following how that requires the mounted hook here (probably just missing something).

Reply:
I think she wanted it to be more magical. Thats one of the points I raised that I wanted your input on :D 

The component tries to find an already existing element with that `id` in the DOM, and if it doesnt, it adds that `id` to itself.

Reply:
I guess I just don't understand the magical part, to be more clear. Why would this logic have to happen after the initial render?

Reply:
So all possible items that may have that ID are rendered? As I said I would probably control this through a prop instead.

Reply:
@mportiz08 I had this logic to check if the document has already an `id` on it where we want to link to. This happens in components that have the id in the beginning of the section itself instead of having it in the title.
Also in this way, we avoid duplication of `id`s in the HTML, since ID attributes must be unique in the document. 

Controlling through a prop, as @dobromir-hristov suggested, would allow us more flexibility but would open the component for possible HTML's errors because we wouldn't prevent the document from duplication of HTML `id`s.

Reply:
I understand why we might choose to sometimes assign the ID attribute to a non-heading element instead of a heading, but I don't understand why we couldn't prevent the ID from being assigned in multiple places in the components themselves—it seems like that is controlled by our app and not outside forces. If that's the case, then we should be able to programmatically decide where the anchor uniquely goes before any `mounted` hook gets called after the initial render.

I think we both agree on the goal of only having unique IDs but maybe there's a little confusion about how to ensure it within the components themselves. Let me know if you want to talk about this offline—I may be missing a subtle thing here that might make this approach necessary.

Reply:
@mportiz08 I set a meeting to continue discussing this. Interesting debate! Thanks :)